Java/serialization/src/main/java/com/amazon/randomcutforest/serialize/json/v1/V1JsonToV2StateConverter.java

```diff
@@ -125,36 +106,98 @@ public PointStoreState getPointStoreState(Precision precision) {
+        checkArgument(serializedForests.size() > 0, "incorrect usage of convert");
+        checkArgument(numberOfTrees > 0, "incorrect parameter");
+        int sum = 0;
+        for (int i = 0; i < serializedForests.size(); i++) {
+            sum += serializedForests.get(i).getNumberOfTrees();
+        }
+        checkArgument(sum >= numberOfTrees, "incorrect parameters");
+
+        RandomCutForestState state = new RandomCutForestState();
+        state.setNumberOfTrees(numberOfTrees);
+        state.setDimensions(serializedForests.get(0).getDimensions());
+        state.setTimeDecay(serializedForests.get(0).getLambda());
+        state.setSampleSize(serializedForests.get(0).getSampleSize());
+        state.setShingleSize(1);
+        state.setCenterOfMassEnabled(serializedForests.get(0).isCenterOfMassEnabled());
+        state.setOutputAfter(serializedForests.get(0).getOutputAfter());
+        state.setStoreSequenceIndexesEnabled(serializedForests.get(0).isStoreSequenceIndexesEnabled());
+        state.setTotalUpdates(serializedForests.get(0).getExecutor().getExecutor().getTotalUpdates());
+        state.setCompact(true);
+        state.setInternalShinglingEnabled(false);
+        state.setBoundingBoxCacheFraction(1.0);
+        state.setSaveSamplerStateEnabled(true);
+        state.setSaveTreeStateEnabled(false);
+        state.setSaveCoordinatorStateEnabled(true);
+        state.setPrecision(precision.name());
+        state.setCompressed(false);
+        state.setPartialTreeState(false);
+
+        ExecutionContext executionContext = new ExecutionContext();
+        executionContext.setParallelExecutionEnabled(serializedForests.get(0).isParallelExecutionEnabled());
+        executionContext.setThreadPoolSize(serializedForests.get(0).getThreadPoolSize());
+        state.setExecutionContext(executionContext);
 
-            CompactSamplerState samplerState = new CompactSamplerState();
-            samplerState.setSize(samples.length);
-            samplerState.setCapacity(sampler.getSampleSize());
-            samplerState.setTimeDecay(sampler.getLambda());
-            samplerState.setPointIndex(pointIndex);
-            samplerState.setWeight(weight);
-            samplerState.setSequenceIndex(sequenceIndex);
-            samplerState.setSequenceIndexOfMostRecentTimeDecayUpdate(0L);
-            samplerState.setMaxSequenceIndex(sampler.getEntriesSeen());
-            samplerState.setInitialAcceptFraction(1.0);
-
-            compactSamplerStates.add(samplerState);
+        SamplerConverter samplerConverter = new SamplerConverter(state.getDimensions(),
+                state.getNumberOfTrees() * state.getSampleSize() + 1, precision, numberOfTrees);
+
+        for (int i = 0; i < serializedForests.size(); i++) {
+            Arrays.stream(serializedForests.get(i).getExecutor().getExecutor().getTreeUpdaters())
+                    .map(V1SerializedRandomCutForest.TreeUpdater::getSampler).forEach(samplerConverter::addSampler);
```

Comment:
suggestion. another way to control the number of operations is to use Stream::limit(n). the way it doesn't iterate through the entire stream with many skips.

Reply:
If you suggest the syntax I am happy to change.

Reply:
yes, it's a suggestion for control the number of adds. a side-benefit is no changes are required for SamplerConverter

Reply:
the suggested code for the for loop is something like this and the changes to SamplerCoverter are not needed:
```
            serializedForests.stream().flatMap(f -> Arrays.steam(f.getExecutor().getExecutor().getTreeUpdaters()))
                .limit(numberOfTrees).map(V1SerializedRandomCutForest.TreeUpdater::getSampler).forEach(samplerConverter::addSampler);
```

Reply:
Done. I think keeping the added check is beneficial in SamplerConverter. If that block of code is executed then the number of trees should not be changed -- because the book keeping in pointstore will be incorrect.